paddle/fluid/pir/transforms/onednn/cpu_bfloat16_placement_pass.cc

```diff
@@ -0,0 +1,594 @@
+                             pir::IrContext *ctx) {
+  auto input_type = type.dyn_cast<IrType1>();
+  return IrType2::get(ctx,
+                      place,
+                      out_dtype,
+                      input_type.dims(),
+                      input_type.data_layout(),
+                      input_type.lod(),
+                      input_type.offset());
+}
+
+template <typename OpType>
+class OneDNNBf16PlacementPattern : public pir::OpRewritePattern<OpType> {
+ public:
+  using pir::OpRewritePattern<OpType>::OpRewritePattern;
+
+  bool MatchAndRewrite(
+      OpType op,
+      pir::PatternRewriter &rewriter) const override {  // NOLINT
+
+    // for (auto& value : op->operands_source()) {
+    //   pir::Type op_dtype = pir::GetDataTypeFromValue(value);
+    //    // Only float input can be converted to bfloat16
+    //   if (!op_dtype.isa<pir::Float32Type>()) {
+    //       return false;
+    //   }
+    // }
+    // The pass use HasOpINT8DataType to skip int8 op
+    auto op_attr = op->attributes();
+
+    if (op_attr.find("mkldnn_data_type") != op_attr.end() &&
+        op_attr.find("mkldnn_data_type")->second == "int8") {
+      return false;
+    }
+    if (op_attr.find("use_quantizer") != op_attr.end() &&
+        op_attr.find("use_quantizer")->second) {
+      return false;
+    }
+
+    std::string target_op_name = op->name();
+    target_op_name.replace(0, 5, "onednn_op");
+
+    auto op_info =
+        pir::IrContext::Instance()->GetRegisteredOpInfo(target_op_name);
+    // pir::IrContext* ctx = pir::IrContext::Instance();
+    if (op_info) {
+      std::vector<pir::Type> op_item_inner_output_types;
+      VLOG(0) << "This is op->num_results:" << op->num_results();
+      for (size_t i = 0; i < op->num_results(); ++i) {
+        // VLOG(0) << "This is result_type:" << op->result_type(i);
+        // pir::Type type = op->result_type(i);
+        // auto out_dtype = type.dyn_cast<pir::DenseTensorType>().dtype();
+        // VLOG(0) << "This is dtype:" << out_dtype;
+        // pir::Type new_type = create_type<pir::DenseTensorType,
+        // paddle::dialect::AllocatedDenseTensorType>( type, phi::CPUPlace(),
+        // pir::BFloat16Type::get(ctx), ctx);
+        // op_item_inner_output_types.push_back(new_type);
+        op_item_inner_output_types.push_back(op->result_type(i));
+      }
+      auto attributes = op->attributes();
+      auto yaml_interface =
+          op_info.GetInterfaceImpl<paddle::dialect::OpYamlInfoInterface>();
+      paddle::dialect::OpRunTimeInfo runtime_info =
+          std::get<3>(yaml_interface->get_op_info_(target_op_name));
+      for (auto &attr : runtime_info.extra_args_default_value) {
+        if (attr.first == "mkldnn_data_type") {
+          VLOG(0) << "mkldnn_data_type set to bf16" << target_op_name;
+          attributes[attr.first] =
+              pir::StrAttribute::get(pir::IrContext::Instance(), "bfloat16");
+        } else {
+          attributes[attr.first] = attr.second;
```

Comment:
Is this necessary? I suppose `attributes[attr.first] == attr.second` is naturally true without any change.